sdk/mobile-center-distribute/src/main/java/com/microsoft/azure/mobile/distribute/Distribute.java

```diff
@@ -785,10 +777,34 @@ else if (Build.VERSION.SDK_INT >= releaseDetails.getMinApiLevel()) {
      * @return true if latest release on server should be used.
      */
     private boolean isMoreRecent(ReleaseDetails releaseDetails) {
+        boolean moreRecent;
         if (releaseDetails.getVersion() == mPackageInfo.versionCode) {
-            return !releaseDetails.getReleaseHash().equals(DistributeUtils.computeReleaseHash(mPackageInfo));
+            moreRecent = !releaseDetails.getReleaseHash().equals(DistributeUtils.computeReleaseHash(mPackageInfo));
+        } else {
+            moreRecent = releaseDetails.getVersion() > mPackageInfo.versionCode;
+        }
+        MobileCenterLog.debug(LOG_TAG, "Latest release more recent=" + moreRecent);
+        return moreRecent;
+    }
+
+    /**
+     * Check if release can be downloaded and installed now.
+     *
+     * @param releaseDetails release.
+     * @return false if release optional AND user has clicked ask me in day since less than 24 hours ago, true otherwise.
+     */
+    private boolean canUpdateNow(ReleaseDetails releaseDetails) {
+        if (releaseDetails.isMandatoryUpdate()) {
+            MobileCenterLog.debug(LOG_TAG, "Release is mandatory, ignoring any postpone action.");
+            return true;
+        }
+        long postponedTime = PreferencesStorage.getLong(PREFERENCE_KEY_POSTPONE_TIME, 0);
+        long postponedUntil = postponedTime + POSTPONE_TIME_THRESHOLD;
+        if (System.currentTimeMillis() < postponedUntil) {
```

Comment:
Can you also add a condition if current time is greater than `postponedTime - POSTPONE_TIME_THRESHOLD`?
There is a scenario that if device time was wrong e.g. 1/1/2020 8:00 AM UTC and it is postponed, it will never prompt the dialog again once the time is back to current unless it is a mandatory update.
In iOS, we are handling this case since this doesn't make any sense. What do you think?